cel/program.go

```diff
@@ -207,6 +207,30 @@ func newProgram(e *Env, ast *Ast, opts []ProgramOption) (Program, error) {
 	if len(p.regexOptimizations) > 0 {
 		decorators = append(decorators, interpreter.CompileRegexConstants(p.regexOptimizations...))
 	}
+	// Enable compile-time checking of syntax/cardinality for string.format calls.
+	if p.evalOpts&OptCheckStringFormat == OptCheckStringFormat {
+		enableCheckArgTypes := ast.IsChecked()
+		isValidType := func(id int64, validTypes ...*types.TypeValue) (bool, error) {
```

Comment:
Does this need to be declared inline? Is there a benefit? Would the code be simpler if it were defined elsewhere?

Reply:
It uses `ast` - we could use a closure in order to continue to use `ast` and define it elsewhere, but I'm not quite sure that's more readable.